### ISSUE TITLE
Oauth login callback not called on Chrome for iOS

### ISSUE THREAD

**Issue by cincue** [google/google-api-javascript-client#272]
I'm integrating Oauth login to an existing web application using as a baseline the javascript code example the [documentation page](https://developers.google.com/identity/sign-in/web/sign-in).

The solution is working fine on every platform save when testing from an iOS Android browser in incognito mode.
The same app works fine in Safari on the same device, both in normal and private mode. We have also tested Chrome on an Android device and it works fine both in normal and incognito mode.

Is this a known problem problem? Do you have any hint on what should I look for?

**Comment by TMSCH** [comment]
HI @cincue,
This is indeed a known problem: Chrome on iOS in Incognito mode disable session/local storage and these are required for the library to function.
We are working on a solution that would allow you to degrade gracefully in such environment, I will keep this thread updated. We are also looking into potential fixes.

**Comment by cincue** [comment]
@TMSCH Thanks for the quick response!! I hope it will be fixed, keep me up to date with this issue please or any emergent fix.

**Comment by TMSCH** [comment]
Hi @cincue, we released a change that allows to handle unsupported environments. Here is the documentation: https://developers.google.com/identity/sign-in/web/reference#googleauththenoninit-onerror. You can now handle environments where Google Sign-In does not work, like Chrome for iOS in Incognito mode.

**Comment by cincue** [comment]
That's great !! It is helpful, thanks a lot.

**Comment by dhruvbalhara** [comment]
@TMSCH  - I used the promise mentioned here https://developers.google.com/identity/sign-in/web/reference#googleauththenoninit-onerror in iOS chrome incognito. I got the response as a success. I am able to open chrome login popup. On successful login, GoogleAuth.signIn() return the error popup_closed_by_user.  Tried this on latest iOS version on chrome 64.  Is this the expected behaviour ?

**Comment by TMSCH** [comment]
@dhruvbalhara  Yes it currently does not work in Chrome on iOS when using incognito.

**Comment by felpin** [comment]
https://developers.google.com/identity/sign-in/web/troubleshooting

The Google sign-in troubleshooting page says that `Google Sign-in is currently not supported in incognito mode on Chrome on iOS.`

**Comment by dvanderb** [comment]
This looks like a question about Google Login and Authentication. Please address this question to the google-signin tag on stackoverflow(https://stackoverflow.com/questions/tagged/google-signin)

If you believe this is in error, please reopen the issue.
